lighthouse-core/lib/sd-validation/schema-validator.js

```diff
@@ -114,20 +123,22 @@ module.exports = function validateSchemaOrg(expandedObj) {
 
   walkObject(expandedObj, (name, value, path, obj) => {
     if (name === TYPE_KEYWORD) {
-      const keyErrorMessages = validateObjectKeys(value, Object.keys(obj));
+      const keyErrors = validateObjectKeys(value, Object.keys(obj));
 
-      keyErrorMessages.forEach(message =>
+      keyErrors.forEach(error => {
         errors.push({
+          invalidTypes: error.invalidTypes,
+          message: error.message,
           // get rid of the first chunk (/@type) as it's the same for all errors
           path:
             '/' +
             path
               .slice(0, -1)
```

Comment:
wait a second, the path is backwards? I think we're missing some tests that have paths of length >2 then 🤔

Reply:
I can look into this more tomorrow, but what makes you think the path is backwards? Here's an example of the path:

```[ 'http://schema.org/author', '0', 'http://schema.org/colleague', '0', '@type' ]```

For this: 

```
{
  "@context": "https://schema.org",
  "@type": "Article",
  "author": {
    "@type": "Person",
    "displayNaame": "Sally",
    "colleague": {
      "@type": "Person",
      "displayNaame": "Sally"
    }
  }
}
```

Reply:
Oh the comment confused me. It says it's getting rid of the "first chunk" but it's removing the last chunk. That made me think it is backwards and all paths started with `/@type`, but I see now :)

maybe update the comment? a test for longer paths still couldn't hurt though sometime :)

Reply:
Oooh, yeah good catch, that's a very confusing comment! Added a test case for deeper nesting too.